clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/ClusteredStore.java

```diff
@@ -803,8 +803,20 @@ public void onAppend(Chain beforeAppend, ByteBuffer appended) {
           }
         }
 
+        public void onAppendInvalidateHash(long hash) {
+          if (clusteredStore.invalidationValve != null) {
+            try {
+              LOGGER.debug("CLIENT: calling invalidation valve for hash {}", hash);
+              clusteredStore.invalidationValve.invalidateAllWithHash(hash);
+            } catch (StoreAccessException sae) {
+              //TODO: what should be done here? delegate to resilience strategy?
```

Comment:
Is there any other choice? And what would the handler be named? invalidateAllWithHashFailure(long hash, StoreAccessException e)

Reply:
Yeah... this issue predates the split. The problem is the failure really needs to be reported back to the original putting thread... but that's on a different client. Depending on the configured consistency it might have left to do other things. I'm not inclined to take fixing this right now.